### ISSUE TITLE
Problem using the file viewer and pressing "i" and having the list be re-sorted

### ISSUE THREAD

**Issue by dataf3l** [vim/vim#5047]
_Instructions: Replace the template text and remove irrelevant text (including this line)_

**Describe the bug**

I have a problem, when using the file viewer and pressing "i" the list of files is re-sorted
This didn't use to be this way in older versions of vim.

**To Reproduce**

1. Run open -a macvim a-folder
2. see the list order
3. press i in normal mode to see file details
4. see how the list is now re-sorted
5. cry, baby, cry
6. find your file again in the list (no, it isn't under the cursor, why would it)
7. get used to the list being re-sorted (this is hard)
8. accept the list is re-sorted (this is harder)
9. be frustrated
10. stop using vim altogether, because, omg.
11. go back to vim, cause you need it
12. file this bug report.

**Expected behavior**
when I press i in the list of files, the file details appear, the list stays in place, in the same order

**Screenshots**
Before:
https://imgur.com/a/FYq1UhK
After:
https://imgur.com/a/vxz8X1L

Notice how the order is different.

**Environment (please complete the following information):**

This doesn't happen in vim, only in macvim:

[b@localhost:~]$ /Applications/MacVim.app/Contents/bin/gvim --version
VIM - Vi IMproved 8.1 (2018 May 18, compiled Jul 28 2019 01:07:30)
macOS version
Included patches: 1-1722
Compiled by travis@Traviss-Mac.local
Huge version with MacVim GUI.  Features included (+) or not (-):
+acl               +file_in_path      +mouse_urxvt       -tcl
+arabic            +find_in_path      +mouse_xterm       +termguicolors
+autocmd           +float             +multi_byte        +terminal
+autochdir         +folding           +multi_lang        +terminfo
-autoservername    -footer            -mzscheme          +termresponse
+balloon_eval      +fork()            +netbeans_intg     +textobjects
+balloon_eval_term +fullscreen        +num64             +textprop
+browse            -gettext           +odbeditor         +timers
++builtin_terms    -hangul_input      +packages          +title
+byte_offset       +iconv             +path_extra        +toolbar
+channel           +insert_expand     +perl/dyn          +transparency
+cindent           +job               +persistent_undo   +user_commands
+clientserver      +jumplist          +postscript        +vartabs
+clipboard         +keymap            +printer           +vertsplit
+cmdline_compl     +lambda            +profile           +virtualedit
+cmdline_hist      +langmap           +python/dyn        +visual
+cmdline_info      +libcall           +python3/dyn       +visualextra
+comments          +linebreak         +quickfix          +viminfo
+conceal           +lispindent        +reltime           +vreplace
+cryptv            +listcmds          +rightleft         +wildignore
+cscope            +localmap          +ruby/dyn          +wildmenu
+cursorbind        +lua/dyn           +scrollbind        +windows
+cursorshape       +menu              +signs             +writebackup
+dialog_con_gui    +mksession         +smartindent       -X11
+diff              +modify_fname      -sound             -xfontset
+digraphs          +mouse             +spell             +xim
+dnd               +mouseshape        +startuptime       -xpm
-ebcdic            +mouse_dec         +statusline        -xsmp
+emacs_tags        -mouse_gpm         -sun_workshop      -xterm_clipboard
+eval              -mouse_jsbterm     +syntax            -xterm_save
+ex_extra          +mouse_netterm     +tag_binary
+extra_search      +mouse_sgr         -tag_old_static
-farsi             -mouse_sysmouse    -tag_any_white
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/Applications/MacVim.app/Contents/Resources/vim"
Compilation: clang -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_MACVIM -Wall -Wno-unknown-pragmas -pipe  -DMACOS_X -DMACOS_X_DARWIN  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: clang   -L. -fstack-protector-strong -L/usr/local/lib -L/usr/local/opt/libyaml/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/readline/lib -L. -fstack-protector-strong -L/usr/local/lib -L/usr/local/opt/libyaml/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/readline/lib  -L/usr/local/lib -o Vim -framework Cocoa -framework Carbon       -lm  -lncurses -liconv -framework AppKit   -fstack-protector  -L/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE

 - OS: macOS 10.14.2
 - Terminal: GUI

**Additional context**
Why vim why...

**Comment by k-takata** [comment]
> This doesn't happen in vim, only in macvim:

Then please report to macvim: https://github.com/macvim-dev/macvim/issues

**Comment by dataf3l** [comment]
Ok, I posted it here, thanks @k-takata  https://github.com/macvim-dev/macvim/issues/954

**Comment by dataf3l** [comment]
Now all I have to do is wait for them to tell me it belongs to Netrw or something like that :)

**Comment by chrisbra** [comment]
If I look at the screenshot this is netrw indeed.

**Comment by eirnym** [comment]
@k-takata Why it's an issue with `MacVim` since there's no difference in runtime files except little of MacVim-specific changes which are definitely not in `netrw`?

**Comment by chrisbra** [comment]
Yes, I think it is a netrw problem. You might try with the latest development snapshot from http://www.drchip.org/astronaut/vim/#NETRW

cc @cecamp
